### PR TITLE
fix(Codespaces): Fixes broken Codespaces Environment & Adds Working Local Dev Container Support

### PR DESCRIPTION
**Pull request by timnolte**
- Updates the Dev Container setup to use a pre-built Dev Container image optimized for WordPress builds.
- Updates the Dev Container setup to allow for proper testing of WordPress Core.
- Reorganizes some of the Dev Container setup as features.
- Updates the Dev Container to properly support using Xdebug without impacting PHPUnit testing.

Trac ticket:  https://core.trac.wordpress.org/ticket/60144

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by kamranzafar4343 — COMMENTED** [review]
looks fine to me, i hope it would be merged

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# gitignore file for WordPress Core
+/bin/bash: line 1: /: Is a directory
```
Comment (swissspidy) [inline .gitignore:1]: ?

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -1,29 +1,68 @@
-version: '3.1'
+version: "3.7"
 
 services:
   app:
-    image: wordpress
+    image: ghcr.io/ndigitals/wp-dev-container:php-8.2-node-20
```
Comment (swissspidy) [inline .devcontainer/docker-compose.yml:5]: Note sure we'd want to use any third-pary image here that's not maintained by WP or Docker 🤔

Comment (timnolte) [reply .devcontainer/docker-compose.yml:5]: There is no prebuilt Docker container that exists that includes everything in this prebuilt Dev Container image that is also built for multi architecture. I have spent time to specifically build this image based off of the WP Dev Docker image. The entire source for it is open for review as well. If you want to force Devs that want to use the Dev Container on Apple chips to have slow subpar, or potentially broken, experiences then you can use the WP Dev image, however it's also going to require significantly longer startup time for Devs as well.

File: .gitignore (hunk 1 above)
Comment (timnolte) [reply .gitignore:1]: Hmm, interesting, that must have been some sort of copy/paste error. I'll get that fixed.

**Comment by anoblet** [comment]
Bump

File: .devcontainer/docker-compose.yml (hunk 2 above)
Comment (peterwilsoncc) [reply .devcontainer/docker-compose.yml:5]: What would need to added to the WordPress Dev contains in order to support this?

**Comment by timnolte** [comment]
I'm closing this PR, I no longer intend to contribute to WordPress.

**Pull request closed without merging**